File: Vinskap.Web/Scripts/app/domain/Bottle.ts
/// <reference path="./Wine.ts" />

class Bottle {

    constructor(public Guid: string, public Wine: Wine, public Year: number, public Price: number, public AddedAt: Date) {
    }

    static fromJson(json: any) {
        return new Bottle(
            json["Guid"],
            Wine.fromJson(json["Wine"]),
            json["Year"],
            json["Price"],
            new Date(Date.parse(json["AddedAt"])));
    }
} 

File: Vinskap.Domain/Bottle.cs
﻿using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Vinskap.Domain
{
    public class Bottle
    {
        public Bottle(Guid guid, Wine wine, int year, double? price, DateTime addedAt)
        {
            Guid = guid;
            Wine = wine;
            Year = year;
            Price = price;
            AddedAt = addedAt;
        }

        public Guid Guid { get; private set; }

        public Wine Wine { get; private set; }

        public int Year { get; private set; }

        public double? Price { get; private set; }

        public DateTime AddedAt { get; private set; }

        public override bool Equals(object obj)
        {
            var other = obj as Bottle;

            if (other == null)
                return false;

            return other.Guid.Equals(this.Guid);
        }

        public override int GetHashCode()
        {
            return Guid.GetHashCode();
        }

    }
}
